src/clj_ldap/client.clj

```diff
@@ -524,9 +535,29 @@
   [pool connection]
   (.releaseAndReAuthenticateConnection pool connection))
 
+(defn bind
+  "Performs a bind operation using the provided connection or pool, bindDN and
+password. If the bind is unsuccessful LDAPException is thrown. Otherwise, a
+map is returned with :code, :name, and optional :diagnostic-message keys.
+The :diagnostic-message might contain password expiration warnings, for instance.
+
+When an LDAP connection object is used as the connection argument the
+bind function will attempt to change the identity of that connection
+to that of the provided DN. Subsequent operations on that connection
+will be done using the bound identity.
+
+If an LDAP connection pool object is passed as the connection argument
+the bind attempt will have no side-effects, leaving the state of the
+underlying connections unchanged."
+  [connection bind-dn password]
+  (let [^BindResult r (basic-bind connection bind-dn password)]
+      (merge (ldap-result r)
```

Comment:
non-standard indentation?

Reply:
Fixed

Reply:
Fixed

Reply:
Clarification: the indentation on the `(merge ,,,)` form is non-standard.

Reply:
Ah, thanks @sumbach. Fixed now along with some other indentation alignments.